supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/GoogleLabels.scala

```diff
@@ -38,8 +41,11 @@ object GoogleLabels {
 
         val foldResult = mainText.toCharArray.foldLeft("")(appendSafe)
 
-        val startsValid = foldResult.headOption.exists(_.isLetter)
-        val endsValid = foldResult.lastOption.exists(_.isLetterOrDigit)
+        val startsValid = if (labelKey) foldResult.headOption.exists(_.isLetter) else foldResult.lastOption.exists(_.isLetterOrDigit)
```

Comment:
The `else` branch looks suspect, it's doing a `lastOption` rather than `headOption` and disallows underscores and dashes.